packages/site/cms-site/src/errorHandler/ErrorHandlerBoundary.tsx

```diff
@@ -0,0 +1,39 @@
+"use client";
+
+import React, { PropsWithChildren } from "react";
+
+type Props = {
+    onError?: (error: Error) => void;
+};
+
+type State = {
+    hasError: boolean;
+};
+
+export class ErrorHandlerBoundary extends React.Component<PropsWithChildren<Props>, State> {
+    constructor(props: Props) {
+        super(props);
+        this.state = { hasError: false };
+    }
+
+    static getDerivedStateFromError() {
+        // Update state so the next render will show nothing instead of the broken block
+        return { hasError: true };
+    }
+
+    componentDidCatch(error: Error) {
```

Comment:
Should we log ErrorInfo (the second argument) here as well? Does this add any helpful information?